crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -227,23 +223,29 @@ impl DoubleEcho {
             return;
         }
 
-        if self.delivered_certificates.get(&cert.id).is_some() {
-            self.event_sender
+        match self.validator_store.get_certificate(&cert.id) {
+            Ok(Some(_)) => self
+                .event_sender
                 .try_send(ProtocolEvents::AlreadyDelivered {
                     certificate_id: cert.id,
                 })
-                .unwrap();
-
-            return;
-        }
-
-        if self
-            .delivery_state_for_new_cert(cert, origin)
-            .await
-            .is_none()
-        {
-            error!("Ill-formed samples");
-            _ = self.event_sender.try_send(ProtocolEvents::Die);
+                .unwrap(),
```

Comment:
I know this didn't change in this PR, but why even use `try_send` is we `unwrap` it right afterwards?